Download size of package.json dependencies

mubaraqwahab opened the pull request:
This should resolve #9 

Here's what I got when testing:

* package.json without dependencies:
```shell
$ node index.js -f package.json
"package.json" dependencies:
None

"package.json" devDependencies:
None

Total size: 0 bytes
```

* package.json with dependencies:
```shell
$ node index.js -f test/package.json
"test/package.json" dependencies:
autoprefixer@9.8.6: 829.58 KiB
parcel-bundler@1.12.4: 14.19 MiB
postcss-modules@3.2.2: 640.74 KiB
tailwindcss@1.7.6: 3.33 MiB

"test/package.json" devDependencies:
stylelint-config-recommended@3.0.0: 2.64 KiB

Total size: 19879005 bytes
```

* package.json with the usual package arguments
```shell
$ node index.js autoprefixer -f test/package.json react react-dom
autoprefixer@9.8.6: 829.58 KiB

"test/package.json" dependencies:
autoprefixer@9.8.6: 829.58 KiB
parcel-bundler@1.12.4: 14.19 MiB
postcss-modules@3.2.2: 640.74 KiB
tailwindcss@1.7.6: 3.33 MiB

"test/package.json" devDependencies:
stylelint-config-recommended@3.0.0: 2.64 KiB

react@16.13.1: 93.23 KiB
react-dom@16.13.1: 794.74 KiB

Total size: 21637770 bytes
```

* Invalid package.json path:
```shell
$ node index.js -f abc/package.json
Error: ENOENT: no such file or directory, open 'abc/package.json'
Total size: 0 bytes
```

* Long option `--file` also works
```shell
$ node index.js --file test/package.json
"test/package.json" dependencies:
autoprefixer@9.8.6: 829.58 KiB
parcel-bundler@1.12.4: 14.19 MiB
postcss-modules@3.2.2: 640.74 KiB
tailwindcss@1.7.6: 3.33 MiB

"test/package.json" devDependencies:
stylelint-config-recommended@3.0.0: 2.64 KiB

Total size: 19879005 bytes
```

**Notes**
1. I changed the option from `-p` in the original issue to `-f` (as in "file"), with the long option `--file`.
2. The total size shows in bytes. Please how do I make it more readable, like "17 KB", "23 MB", etc.?

Also, if there's anything I didn't do right, please let me know. Thank you @arve0.

mubaraqwahab committed:
Destructure argv from process

mubaraqwahab committed:
Get package.json path from argv

mubaraqwahab committed:
read package.json works

mubaraqwahab committed:
Transform dependencies objects to arrays

mubaraqwahab committed:
Log sizes of package.json dependencies

mubaraqwahab committed:
Factor out getPackageSize

mubaraqwahab committed:
Log sizes of devdeps too, and regular args

mubaraqwahab committed:
Add total size in bytes

mubaraqwahab committed:
Add newlines to output for clarity

mubaraqwahab committed:
Add newline before logging devdeps

mubaraqwahab committed:
Update the test script

arve0 commented:
Looks good👍

> The total size shows in bytes. Please how do I make it more readable, like "17 KB", "23 MB", etc.?

On the [server side](https://github.com/arve0/npm-download-size-api/blob/master/resolve.ts#L146:L153), I use [si-prefix](https://www.npmjs.com/package/si-prefix).

mubaraqwahab commented:
Alright thank you.

mubaraqwahab commented:
I'm guessing the [Usage message](https://github.com/arve0/npm-download-size-cli/blob/fb65e945529d51c6e5543d5eedc4e5705413f047/index.js#L18) should change as well. What should it be now?

arve0 committed:
refactor, total sizes

- get all packages first
- outer catch
- print with prefix when package file
- show complete sizes in kilo, mega, etc
- some refactoring
